src/twisted/names/authority.py

```diff
@@ -302,17 +303,17 @@ class BindAuthority(FileAuthority):
     Supports only C{$ORIGIN} and C{$TTL} directives.
     """
 
-    def loadFile(self, filename):
+    def loadFile(self, filename: AnyStr) -> None:
```

Comment:
```suggestion
    def loadFile(self, filename: bytes | str) -> None:
```

Reply:
This `bytes | str ` causes tests to fail,  ```builtins.TypeError: unsupported operand type(s) for |: 'type' and 'type'``` seems python3.8 does not support the pipe operator.

Reply:
FYI the way we usually fix these is not to use `typing.Union`, since 3.8 support will be dropped pretty soon now. If you add `from __future__ import annotations` to the top of the file, that prevents the annotations from being evaluated at import time (which is faster for newer versions of Python anyway)